examples/tv-casting-app/APIs.md

```diff
@@ -0,0 +1,692 @@
+for its respective platform to consume any of the APIs described in this
+document. Refer to the tv-casting-app READMEs for [Android](android) and
+[iOS](darwin) to understand how to build and consume each platform's specicific
+libraries.
+
+### Initialize the Casting Client
+
+_{Complete Initialization examples:
+[Android](android/App/app/src/main/java/com/matter/casting/InitializationExample.java)
+| [iOS](darwin/TvCasting/TvCasting/TvCastingApp.swift)}_
+
+A Casting Client must first initialize the Matter SDK and define the following
+[`DataProvider`](android/App/app/src/main/jni/com/matter/casting/params/DataProvider.java)
+objects for the the Matter Casting library to use throughout the client's
+lifecycle:
+
+1.  **Rotating Device Identifier** - Refer to the Matter specification for
+    details on how to generate the
+    [Rotating Device Identifier](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/rendezvous/DeviceDiscovery.adoc#245-rotating-device-identifier)).
+    Then, instantiate a `DataProvider` object as described below.
+
+    On Android, define a `rotatingDeviceIdUniqueIdProvider` to provide the
+    Casting Client's RotatingDeviceIdUniqueId, by implementing a `DataSource`:
+
+    ```java
+    private final static DataProvider<byte[]> rotatingDeviceIdUniqueIdProvider = new DataProvider<byte[]>() {
+        private static final String ROTATING_DEVICE_ID_UNIQUE_ID = "EXAMPLE_ID";    // dummy value for demonstration only
+        @Override
+        public byte[] get() {
+            return APP_ID.getBytes();
+        }
+    }
+    ```
+
+    On iOS, define the
+    `func castingAppDidReceiveRequestForRotatingDeviceIdUniqueId` in a class,
+    `AppParametersDataSource`, that implements the `MTRDataSource`:
+
+    ```objectivec
+    class AppParametersDataSource : NSObject, MTRDataSource
+    {
+        func castingAppDidReceiveRequestForRotatingDeviceIdUniqueId(_ sender: Any) -> Data {
+            return "YOUR_GENERATED_ROTATING_DEVICE_ID".data(using: .utf8)!      // dummy value for demonstration only
+        }
+        ...
+    }
+    ```
+
+2.  **Commissioning Data** - This object contains the passcode, discriminator,
+    etc which identify the app and are provided to the CastingPlayer during the
+    commissioning process. Refer to the Matter specification's
+    [Onboarding Payload](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/qr_code/OnboardingPayload.adoc#ref_OnboardingPayload))
+    section for details on commissioning data.
+
+    On Android, define a `commissioningDataProvider` that can provide the
+    required values to the CastingApp.
+
+    ```java
+    private final static DataProvider<CommissioningData> commissioningDataProvider = () -> {
+        CommissioningData commissioningData = new CommissioningData();
+        commissioningData.setSetupPasscode(20202021);   // dummy values for demonstration only
+        commissioningData.setDiscriminator(3874);
```

Comment:
this also is a dummy value for demonstration only

Reply:
Updated in project-chip#29444